src/OSPSuite.Infrastructure.Import/Services/SimulationPKAnalysesImporter.cs

```diff
@@ -101,34 +113,56 @@ private double convertValueToCoreValue(IDimension dimension, double valueInUnit,
          return dimension.UnitValueToBaseUnitValue(unit, valueInUnit);
       }
 
-      private QuantityPKParameter retrieveOrCreatePKParameterFor(CsvReader csv)
+      private QuantityPKParameter retrieveOrCreatePKParameterFor(CsvReader csv, IModelCoreSimulation modelCoreSimulation)
       {
          var parameterName = csv[PARAMETER_NAME];
          var quantityPath = csv[QUANTITY_PATH];
          var id = QuantityPKParameter.CreateId(quantityPath, parameterName);
-         if (!_importedPK.Contains(id))
-         {
-            var dimension = findDimensionFor(csv[UNIT]);
-            var pkParameter = new QuantityPKParameter {Name = parameterName, QuantityPath = quantityPath, Dimension = dimension};
-            _importedPK.Add(pkParameter);
-         }
+         if (_importedPK.Contains(id))
+            return _importedPK[id];
+
+
+         var dimension = findDimensionFor(parameterName, csv[UNIT]);
+         var pkParameter = new QuantityPKParameter {Name = parameterName, QuantityPath = quantityPath, Dimension = dimension};
+         var quantityPKParameterContext = new QuantityPKParameterContext(pkParameter, modelCoreSimulation.MolWeightFor(quantityPath));
+         pkParameter.Dimension = _dimensionFactory.MergedDimensionFor(quantityPKParameterContext);
+         _importedPK.Add(pkParameter);
 
          return _importedPK[id];
       }
 
-      private IDimension findDimensionFor(string unit)
+      private IDimension findDimensionFor(string parameterName, string unit)
       {
          if (string.IsNullOrEmpty(unit))
             return _dimensionFactory.NoDimension;
 
 
          var dimension = _dimensionFactory.DimensionForUnit(unit);
 
-         if (dimension != null)
-            return dimension;
+         if (dimension == null)
+         {
+            _logger.AddWarning(Error.CouldNotFindDimensionWithUnit(unit));
+            return _dimensionFactory.CreateUserDefinedDimension(parameterName, unit);
+         }
 
+         // for a mass dimension, we will return the Molar dimension to ensure that we return the value of the PK-Parameter in the base supported unit
```

Comment:
We always want the SOURCE DIMENSION of a merge dimension to be molar. Otherwise the base unit would be mg, thus mixing and matching